forte/data/readers/corpus_reader.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2019 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+A reader to read passages from `MS MARCO` dataset, pertaining to the
+Passage Ranking task. Uses the document text for indexing.
+
+Official webpage -
+https://github.com/microsoft/MSMARCO-Passage-Ranking#data-information-and-formating
+Dataset download link -
+https://msmarco.blob.core.windows.net/msmarcoranking/collection.tar.gz
+Dataset Paper -
+Nguyen, Tri, et al. "MS MARCO: A Human-Generated MAchine Reading
+COmprehension Dataset." (2016).
+"""
+import os
+from typing import Iterator, Tuple
+
+from texar.torch import HParams
+
+from ft.onto.base_ontology import Document
+from forte.data import DataPack
+from forte.data.readers.base_reader import PackReader
+from forte.common.resources import Resources
+
+__all__ = [
+    "CorpusReader"
+]
+
+
+class CorpusReader(PackReader):
```

Comment:
Apparently, this class name, and the filename are too generic. Isn't this for MS-Macro only?